theano/gof/op.py

```diff
@@ -783,6 +783,9 @@ def __new__(cls, *args, **kwargs):
         # this function exists to silently and transparently ensure that all
         # existing Ops get a _op_use_c_code attribute
         obj = object.__new__(cls)
+        # Ensuring all the Ops created henceforth has __props__
+        if not hasattr(obj, '__props__'):
+            raise
```

Comment:
Just to be sure, do you mean we can't cache any ops with a metaclass without having all nodes implement props or you mean we will only cache ops with props?

The second is fine.